src/libraries/System.Private.CoreLib/src/System/Resources/ResourceReader.cs

```diff
@@ -917,40 +920,51 @@ private void _ReadResources()
         // and initialize Reflection.
         private Type FindType(int typeIndex)
         {
+            if (!AllowCustomResourceTypes)
+            {
+                throw new NotSupportedException(SR.ResourceManager_ReflectionNotAllowed);
+            }
+
             if (typeIndex < 0 || typeIndex >= _typeTable.Length)
             {
                 throw new BadImageFormatException(SR.BadImageFormat_InvalidType);
             }
-            if (_typeTable[typeIndex] == null)
+
+            return _typeTable[typeIndex] ?? UseReflectionToGetType(typeIndex);
+        }
+
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2057:TypeGetType",
```

Comment:
I kind of think this method should have `RequiresUnreferencedCode` on it. And then above in `FindType` we suppress the message.

The reasoning being if some other call to `UseReflectionToGetType` gets added, they will get a warning. Only the `FindType` ensures that the unreferenced code isn't called by throwing on `!AllowCustomResourceTypes`